bitbucket-pipelines.yml

```diff
@@ -0,0 +1,91 @@
+# Define the Docker image to use for all steps
+image: 
+  name: aaronmedinaaligent/shopify-cli-base:latest
+  username: $DOCKER_HUB_USERNAME
+  password: $DOCKER_HUB_PASSWORD
+
+definitions:
+  steps:
+    - step: &theme-check
+        name: Theme Check
+        services:
+          - docker
+        caches:
+          - docker
+        script:
+          - echo "STARTING THEME CHECK..."
+          - echo "Verifying Ruby, Node.js, and Shopify CLI installations..."
+          - ruby -v      
+          - node -v     
+          - npm -v
+          - shopify version  
+          - ls -al        
+          - echo "Checking if /theme directory exists..."
+          - if [ ! -d "./theme" ]; then echo "/theme directory does not exist."; exit 1; fi
+          - echo "Listing files in the /theme directory..."
+          - find theme -exec ls -ld --time-style=long-iso {} \; | awk '{print $6, $7, $8, $9}'
+          - echo "Running shopify theme check --fail-level=$FAIL_LEVEL --path=theme"
+          - shopify theme check --fail-level=$FAIL_LEVEL --path=theme
+
+    - step: &deploy
+        name: Deploy Theme
+        services:
+          - docker
+        caches:
+          - docker
+        script:
+          - echo "Deploying theme..."
+          - ruby -v
+          - node -v
+          - npm -v
+          - shopify version
+          - ls -al
+          - if [ ! -d "./theme" ]; then echo "/theme directory does not exist."; exit 1; fi
+          - find theme -exec ls -ld --time-style=long-iso {} \; | awk '{print $6, $7, $8, $9}'
+          - |
+            if [ "$BITBUCKET_DEPLOYMENT_ENVIRONMENT" == "staging" ]; then
+              echo "Running shopify theme check --fail-level=$FAIL_LEVEL --path=theme"
+              shopify theme check --fail-level=$FAIL_LEVEL --path=theme
+              echo "Deploying to..."
+              echo -e "  _____ _______       _____ _____ _   _  _____ \n / ____|__   __|/\   / ____|_   _| \ | |/ ____|\n| (___    | |  /  \ | |  __  | | |  \| | |  __ \n \___ \   | | / /\ \| | |_ | | | | .   | | |_ |\n ____) |  | |/ ____ \ |__| |_| |_| |\  | |__| |\n|_____/   |_/_/    \_\_____|_____|_| \_|\_____|\n"
```

Comment:
could this be simplified? the only difference seem to be the `echo` (which could be removed for readability) and the `--no-delete` vs `--allow-live` for production

Reply:
I have modified the yml file and added the arguments to bitbucket's deployment variable